pkg/operator/certrotation/signer_test.go

```diff
@@ -276,7 +276,7 @@ func TestEnsureSigningCertKeyPair(t *testing.T) {
 			verifyActions: func(t *testing.T, updateOnly bool, client *kubefake.Clientset, controllerUpdatedSecret bool) {
 				t.Helper()
 				actions := client.Actions()
-				if len(actions) != 2 {
+				if len(actions) != 4 {
```

Comment:
why does the create scenario sends additional `get` and `update` requests?

Reply:
Previously only one `create` was invoked, because `ensureMetadataUpdate` returned true (no ownership annotations/refs) and `ensureSecretTLSTypeSet` returned `false` (if ResourceVersion is unset no type change necessary). That means the secret was created after `needNewSigningCertKeyPair` was returning `true`.

With this change `needsMetadataUpdate || needsTypeUpdate` and secret is created with empty data and required annotaitons/ownerrefs, later being updated with secret contents when `needNewSigningCertKeyPair` returns `true`.

This may seem like extra actions, but it protects us from unexpected changes when secret contents and metadata are fully compliant, but secret type is still invalid. Its not possible in the standard flow for born-in-4.7-clusters, but may be possible in later cycles